spark/src/main/java/org/apache/iceberg/actions/RemoveOrphanFilesAction.java

```diff
@@ -161,15 +172,44 @@ public RemoveOrphanFilesAction deleteWith(Consumer<String> newDeleteFunc) {
         .as(Encoders.STRING())
         .collectAsList();
 
-    Tasks.foreach(orphanFiles)
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc))
-        .run(deleteFunc::accept);
+    paralleExecutor(sparkContext, orphanFiles, excutorTaskNum, deleteFunc);
 
     return orphanFiles;
   }
 
+  private void paralleExecutor(JavaSparkContext javaSc, List<String> orphanFiles,
+                              int numSlices, Consumer deleteConsumer) {
+    if (numSlices == 0) {
+      Tasks.foreach(orphanFiles)
+          .noRetry()
+          .suppressFailureWhenFinished()
+          .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc))
+          .run(deleteConsumer::accept);
+    } else {
+      javaSc.parallelize(orphanFiles, numSlices).foreachPartition(row -> {
+        List<String> orphanFileList = IteratorUtils.toList(row);
+        Tasks.foreach(orphanFileList)
+            .noRetry()
+            .suppressFailureWhenFinished()
+            .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc))
+            .run(deleteConsumer::accept);
+      });
+    }
+  }
+
+  public static class DeleteConsumer implements Consumer<String>, Serializable {
```

Comment:
I wonder if it makes sense to add a Serializable Consumer class as a utility. Then have this DeleteConsumer extend that.